scipy/optimize/optimize.py

```diff
@@ -693,7 +693,15 @@ def _approx_fprime_helper(xk, f, epsilon, args=(), f0=None):
     for k in range(len(xk)):
         ei[k] = 1.0
         d = epsilon * ei
-        grad[k] = (f(*((xk + d,) + args)) - f0) / d[k]
+        df = (f(*((xk + d,) + args)) - f0) / d[k]
+        if not np.isscalar(df):
```

Comment:
As discussed in https://github.com/scipy/scipy/pull/10673#discussion_r352371584, `isscalar` is not usually the best approach.